adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m

```diff
@@ -13,47 +13,90 @@
+    if (appIds.count > 1) {
+        GADMPangleLog(@"Found the following App IDs:%@. Please remove any app IDs which you are not using", appIds);
+        GADMPangleLog(@"Configuring Pangle SDK with the app ID:%@", appID);
+    }
+    
+    BUAdSDKConfiguration *sdkConfiguration = [BUAdSDKConfiguration configuration];
+    sdkConfiguration.territory = BUAdSDKTerritory_NO_CN;
+    sdkConfiguration.appID = appID;
+    sdkConfiguration.coppa = @(_coppa);
+    sdkConfiguration.GDPR = @(_gdpr);
+    sdkConfiguration.CCPA = @(_ccpa);
+    [BUAdSDKManager startWithAsyncCompletionHandler:^(BOOL success, NSError *error) {
+        completionHandler(error);
+    }];
 }
 
 + (GADVersionNumber)adSDKVersion {
-  // TODO: Populate `versionString` with the ad-network's SDK version in NSString format.
-  NSString *versionString = @"";
-  GADVersionNumber version = {0};
-  NSArray<NSString *> *components = [versionString componentsSeparatedByString:@"."];
-  if (components.count == 3) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue;
-  } else {
-    NSLog(@"Unexpected ad SDK version string: %@. Returning 0 for adSDKVersion.", versionString);
-  }
-  return version;
+    NSString *versionString = BUAdSDKManager.SDKVersion;
+    NSArray *versionComponents = [versionString componentsSeparatedByString:@"."];
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 4) {
+        version.majorVersion = [versionComponents[0] integerValue];
+        version.minorVersion = [versionComponents[1] integerValue];
+        version.patchVersion = [versionComponents[2] integerValue] * 100 + [versionComponents[3] integerValue];
+    }else {
+        GADMPangleLog(@"Unexpected version string: %@. Returning 0 for adSDKVersion.", versionString);
+    }
+    return version;
 }
 
 + (GADVersionNumber)adapterVersion {
-  GADVersionNumber version = {0};
-  NSArray<NSString *> *components = [GADMAdapterPangleVersion componentsSeparatedByString:@"."];
-  if (components.count == 4) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue * 100 + components[3].integerValue;
-  }
-  return version;
+    GADVersionNumber version = {0};
+    NSArray<NSString *> *components = [GADMAdapterPangleVersion componentsSeparatedByString:@"."];
+    if (components.count >= 4) {
+        version.majorVersion = components[0].integerValue;
+        version.minorVersion = components[1].integerValue;
+        version.patchVersion = components[2].integerValue * 100 + components[3].integerValue;
+        if (components.count >= 5) {
+            version.patchVersion = version.patchVersion * 100 + components[4].integerValue;
+        }
+    }
+    return version;
 }
 
 + (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
```

Comment:
Remove `TODO` if there isn't a class from Pangle that passes in mediation extras.

Reply:
`-networkExtrasClass` is `GADMediationAdapter`protocol require method，can I remove?

Reply:
`networkExtrasClass` is `GADMediationAdapter`protocol require method，can I remove?

Reply:
I am referring to the `// TODO: Return the class for passng in mediation extras (if any). Else, return `Nil`.` comment below, thanks!

Reply:
ok~